### ISSUE TITLE
Counterintuitive looping controls

### ISSUE THREAD

**Issue by lookitsnicholas** [LMMS/lmms#6014]
relates to #4034 and #5505 - not really sure why this needs to be separate, tbh

### Bug Summary

As explained [here](https://github.com/LMMS/lmms/pull/4034#issuecomment-833164078), many people have expressed issues with the new control scheme for creating loops because it implicitly discourages changing parameters for larger projects due to the new additional amount of input required in order to do so.

#### Steps to reproduce

Create a loop of 100 measures, and try to move the loop end point to measure 50. Additionally, attempt to create a four-bar loop and move the end of the loop to the start of measure two.

#### Expected behavior

Something analogous to the original behavior, with one right-click/two-finger tap to end the loop and Shift+(right-click/two-finger tap) to start the loop.

#### Actual behavior

Weirdness due how the mean calculations are implemented... the video below shows the inability to end the loop at measure 50, and how incremental decrements of the loop length are currently required to shrink a four-bar loop into one. Zooming out is not an acceptable workaround because the resolution of said changes drops significantly, resulting in potentially more needed clicks.

#### Recording

https://user-images.githubusercontent.com/31457277/117765091-743d1580-b1fb-11eb-8594-7a2c58082df1.mov

#### Affected LMMS versions

Anything recent on 1.3-alpha

#### Feedback

My input on this is that this change is breaking mine and others' workflows because of the unnecessary amount of time clicking around and zooming in order to achieve the same result. A suggestion would be to either revert this behavior, add an option to toggle to eliminate extra button presses to accommodate the original behavior, or at the very least the addition of Shift modifiers (Shift+LMB for the start of a loop and Shift+RMB for the end of a loop) to get rid of the movement now required to quickly change loop parameters.

**Comment by oldcastlehq** [comment]
I'm second to this. It's not only unintuitive, but it makes your work on LMMS harder.

**Comment by Spekular** [comment]
> incremental decrements of the loop length are currently required to shrink a four-bar loop into one.

For the record this is false. Simply click anywhere on the right half to select the endpoint and then drag it left until the loop is one bar long.

**Comment by lookitsnicholas** [comment]
Technically...? On a MacBook, that would consist of zooming out by scrolling on the zoom list box + Control+click + dragging the loop end point to the desired destination - that does not necessarily address the left point in that instance, which would also require zooming out. The point remains that that extra step of zooming out is now suddenly required when the previous implementation/other DAWs would allow for at most two-click operation.

**Comment by GingkathFox** [comment]
> Technically...? On a MacBook, that would consist of zooming out by scrolling on the zoom list box + Control+click + dragging the loop end point to the desired destination - that does not necessarily address the left point in that instance, which would also require zooming out. The point remains that that extra step of zooming out is now suddenly required when the previous implementation/other DAWs would allow for at most two-click operation.

I agree, I have to move back and forth in the piano roll to move the loop when I wanna loop a specific bar later in the song. It's annoying at best, and it really sucks.

**Comment by Xeno-Idaltu** [comment]
Its been more than a year and still no changes to this annoying feature.. -__-*

**Comment by RiedleroD** [comment]
the devs are almost done with the reorg. Have patience, young padawan.

**Comment by RoxasKH** [comment]
Honestly even if there are people complaining that doesn't there aren't people who like it. I'd personally focus on making better the new system rather than purely revert the PR.

In my point of view the new system is way more intuitive to the newbie users, and other daws use the same system as well.

This is just for saying that probably a poll would be needed, as it's clear that people that don't like it would complain, but people that actually like the new system won't even think about saying somwhere they like it.

That said, i know it can be less comfortable in some situations, but it can also be more comfortable in some other situations, it just requires to change the workflow you're used to, and it shouldn't be so difficult as we learned how to use a daw from zero (and if you use multiple daws, you'd also be more flexible).
Now i'm not saying you aren't allowed to not like it obviously, i'm just saying that as it's present only in the alpha build, not even all the userbase have had the chance to try the new system (and i'm one of them, still sticking on 1.2.0), and that a lot of people are complaining but it doesn't really immediately imply it's bad and only bad.

Also i read some terrible and hilarious comment about this PR being made only due to the author's ego, while people should know that lmms is one of the most cautios and slow project to merge a PR, and that it wouldn't have been merged if there weren't good reasons behind them. As i said, it's used in other daws as well, and it's more intuitive as it doesn't require to know which keyboard key you need to use with (i personally needed to search online to get how to set the start loop).

This is just to balance things, and i personally think a compromise it's needed as well. For example it could be kept as the new one, while leaving the ability to set the starting point holding shift. This would half the problems shown.

**Comment by RiedleroD** [comment]
TL;DR: Let's maybe open a discussion regarding this? Github discussions exist for situations like these. Would be best if someone that uses LMMS a lot could open the discussion with a few Pro/Con of several variants.

I personally like the idea of pressing shift to set the starting point and setting the ending point otherwise. It's very quick and imo also intuitive.

**Comment by RoxasKH** [comment]
Tbh i can't see how someone that start sentences with TLDR would be open to discuss.

Also i think this issue was already opened a few times, duplicates just get closed, so that discussion is kept only in one place.

**Comment by JohannesLorenz** [comment]
Duplicate to #5505 . Let's close this?

**Comment by Spekular** [comment]
> Duplicate to #5505 . Let's close this?

Indeed.
